flink: 1.5 -> 1.6

mbode opened the pull request:
###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

mbode committed:
flink: 1.5.0 -> 1.6.0

mbode committed:
flink_1_5: 1.5.0 -> 1.5.3

xeji commented:
Why do we need to keep 4 versions of this in nixpkgs? 
According to upstream https://flink.apache.org/downloads.html#update-policy-for-old-releases :

> As of March 2017, the Flink community decided to support the current and previous minor release with bugfixes. If 1.2.x is the current release, 1.1.y is the previous minor supported release. Both versions will receive bugfixes for critical issues.

So IMO we should keep 1.5.3 and 1.6.0, and drop the older versions.

mbode committed:
flink: drop older versions than current and previous minor release

This is consistent with the update policy as laid out in https://flink.apache.org/downloads.html#update-policy-for-old-releases.
